core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -112,6 +131,33 @@ class DefaultAutoTopicCreationManager(
     }
   }
 
+  override def getTopicCreationErrors(
+    topicNames: Set[String]
+  ): Map[String, String] = {
+    val currentTime = time.milliseconds()
+    val errors = mutable.Map.empty[String, String]
+    val expiredKeys = mutable.Set.empty[String]
+    
+    // Check requested topics and collect expired keys
+    topicNames.foreach { topicName =>
+      Option(topicCreationErrorCache.get(topicName)) match {
+        case Some(cachedError) if (currentTime - cachedError.timestamp) <= errorCacheTtlMs =>
+          errors.put(topicName, cachedError.errorMessage)
+        case Some(_) =>
+          expiredKeys += topicName
+        case None =>
+      }
+    }
+    
+    // Remove expired entries
+    expiredKeys.foreach { key =>
+      topicCreationErrorCache.remove(key)
+      debug(s"Removed expired topic creation error cache entry for $key")
+    }
+    
+    errors.toMap
+  }
+
   private def sendCreateTopicRequest(
```

Comment:
`sendCreateTopicRequest` is used by both streams and other internal topics. I would suggest making a separate copy for this method to not affect the code path for other internal topics (since the error message will not be used on those code paths.)